File: Binary_Search/Ruby/jcla1/binary_search_test.rb
require './binary_search'

describe "#binary_search" do
  it "Test empty array" do
    binary_search([], 1).should eq(-1)
  end

  it "Search last key" do
    binary_search([1,2,3,4,5], 5).should eq(4)
  end

  it "Search first key" do
    binary_search([1,2,3,4,5], 1).should eq(0)
  end

  it "Search middle key" do
    binary_search([1,2,3,4,5], 3).should eq(2)
  end

  it "Search not found key" do
    binary_search([1,2,3,4,5], 10).should eq(-1)
  end
end

File: Binary_Search/Scala/aayushKumarJarvis/BinarySearchTest.scala
import org.scalatest.{Matchers, ShouldMatchers, FunSuite}

class BinarySearchTest extends FunSuite with Matchers {

  test("Should search for") {

    val objectForBinarySearch = new BinarySearch

    objectForBinarySearch.binarySearch(Array(1,2,3,4,5),3) should be(2)
    objectForBinarySearch.binarySearch(Array(1,2,3,4,5),4) should be(3)
    objectForBinarySearch.binarySearch(Array(0,0,0,1,1,1),1) should be(3)
  }
}
